knowledge-base/common-change-selected-localization-keys.md

```diff
@@ -0,0 +1,197 @@
+    builder.Services.AddSingleton<WeatherForecastService>();
+
+    // register a custom localizer for the Telerik components, after registering the Telerik services
+    builder.Services.AddSingleton(typeof(ITelerikStringLocalizer), typeof(ResxLocalizer));
+
+    #region Localization
+
+    builder.Services.AddControllers();
+    builder.Services.AddLocalization(options => options.ResourcesPath = "Resources");
+    builder.Services.Configure<RequestLocalizationOptions>(options =>
+    {
+        // define the list of cultures your app will support
+        var supportedCultures = new List<CultureInfo>()
+        {
+            new CultureInfo("en-US")
+        };
+
+        options.SupportedCultures = supportedCultures;
+        options.SupportedUICultures = supportedCultures;
+    });
+
+    // the custom localizer service is registered later, after the Telerik services
+
+    #endregion
+
+    var app = builder.Build();
+
+    // Configure the HTTP request pipeline.
+    if (!app.Environment.IsDevelopment())
+    {
+        app.UseExceptionHandler("/Error");
+        // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
+        app.UseHsts();
+    }
+
+    #region Localization
+
+    app.UseRequestLocalization(app.Services.GetService<IOptions<RequestLocalizationOptions>>().Value);
+
+    #endregion
+
+    app.UseHttpsRedirection();
+
+    app.UseStaticFiles();
+
+    app.UseRouting();
+
+    app.MapControllers();
+    app.MapBlazorHub();
+    app.MapFallbackToPage("/_Host");
+
+    app.Run();
+    ````
+3. Add a cookie in the `_Host.cshtml` file to store the culture choice:
+    >caption _Host.cshtml
+    ````CSHTML
+    @using Microsoft.AspNetCore.Localization
+    @using System.Globalization
+
+    . . . .
+    <body class="k-body">
+
+        @* Culture cookie start *@
+
+        @{
+            this.HttpContext.Response.Cookies.Append(CookieRequestCultureProvider.DefaultCookieName,
+                CookieRequestCultureProvider.MakeCookieValue(new RequestCulture(CultureInfo.CurrentCulture, CultureInfo.CurrentUICulture))
+            );
+        }
+
+        @* Culture cookie end *@
+
+        <app>
+            <component type="typeof(App)" render-mode="ServerPrerendered" />
+        </app>
+
+    . . . .
+
+    </body>
+    ````
+4. Create a services folder and implement the Telerik localization service with a list of all needed localization keys and their corresponding custom strings. The service also relies on a `~/Resources` folder with the necessary `.resx` files. You can find an up-to-date list of the used strings in our [API documentation](https://docs.telerik.com/blazor-ui/api/Telerik.Blazor.Resources.Messages) and the [offline version](https://www.telerik.com/account/my-downloads) of our [Demo solution](https://demos.telerik.com/blazor-ui). Telerik maintains the main `TelerikMessages.resx` file (in English) to have the full list with each release.
```

Comment:
It makes more sense to move step 4 before step 1. Currently the app is broken between 1 and 4, because ResxLocalizer is undefined.